bitcoin-core-dev-tech/2018-03-06-taproot-graftroot-etc.es.md

```diff
@@ -0,0 +1,124 @@
+
+# Volver a graftroot
+
+Primero hago mi script. Luego hago una firma aleatoria... un hash de un... y luego de ese par de script de firma, calculo la pubkey. Y luego establezco esa pubkey. Puedo mostrarte que la-- la preimagen para esta firma, así que esto es básicamente probarte que.... no hay... que pueda ser posible. podrías hacer más de una.
+
+....
+
+Dados dos mensajes y una sola firma fija, encuéntrame una clave pública... Dados dos mensajes, encuéntrame una clave pública que firme ambos, a no ser que conozcas la... Creo que tengo una prueba informal de que..
+
+...
+
+Probablemente nunca deberías enviar dinero a personas que no esperan recibirlo. Pero la gente probablemente seguirá haciéndolo de todos modos.
+
+Siempre hay una firma global en la transacción. Hay una firma en alguna parte de la transacción.  Todo se agrega a la firma. ...
+
+Un graftroot donde se puede enviar de forma no interactiva al multisig de alguien... a algún script arbitrario para el multisig con claves en él. Podrías hacer, pero haces el tihng directo. Quiero un script multisig de `2` de `3` y quiero hacerlo sin pedirle a la gente que firme nada ni hacer ningún injerto. No se puede. ¿Estás seguro? Puedes, con el recovery tricky, siempre que haya al menos un único script. Puede que no funcione aunque.... esta firma no firma el txid, sólo firma los scripts. Funciona. Pero sólo mientras haya un solo... y si estás haciendo eso, ¿por qué no estás usando taproot? Es más barato. La única razón por la que lo estás haciendo es porque quieres graduarte de taproot a graftroot... por razones de fungibilidad tú.. ¿No sería genial fusionarlo todo? Digamos que yo tengo una clave pública, y tengo un script, y arreglado por adelantado, y tú no puedes. Usted quiere hacer graftroot. Si `P` es fijo, entonces usted no puede. Pero si sólo quieres enviar a un script, entonces podrías usar graftroot. Quieres enviar de forma no interactiva a un script... queremos enviar de forma no interactiva a un script que no tiene relación con esta pubkey con la que no tiene relación, pero eso es imposible. Dices que es inusual que tengas una clave pública y un script por adelantado. Si no tienes no-interactividad, entonces graftroot hace todo. `2` de `3` multisig-- `3` posibles scripts ahora. Eso no es tan eficiente como tener `3` scripts.. entonces de nuevo, `2`-de-`3`... En graftroot, deberías estar enviando a una colección de pubkeys, y ellos pueden decidir delegar a scripts particulares. Mientras tengas esta colección de llaves públicas, es completamente no interactivo.
+
+...
+
+# Taproot and graftroot
+
+Taproot: `P = c + H(c || script) G`
+
+Graftroot: `sig<sub>p</sub>(script)`
+
+Una salida es un punto `p`. `c` es algún otro punto. Hay algún otro punto, y su clave privada puede no existir.
+
+El gasto directo es `sig<sub>p</sub>(tx).`
```

Comment:
same here
```suggestion
El gasto directo es sig<sub>p</sub>(tx).
```

Reply:
Okay, thank you, I'll correct it later. :)